oauth2/oauth2_auth_code_test.go

```diff
@@ -917,12 +917,22 @@ func TestAuthCodeWithMockStrategy(t *testing.T) {
 						hs := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 							assert.Equal(t, r.Header.Get("Content-Type"), "application/json; charset=UTF-8")
 
+							expectedGrantedScopes := []string{"openid", "offline", "hydra.*"}
+							expectedSubject := "foo"
+
 							var hookReq hydraoauth2.RefreshTokenHookRequest
 							require.NoError(t, json.NewDecoder(r.Body).Decode(&hookReq))
-							require.Equal(t, hookReq.Subject, "foo")
-							require.ElementsMatch(t, hookReq.GrantedScopes, []string{"openid", "offline", "hydra.*"})
+							require.Equal(t, hookReq.Subject, expectedSubject)
+							require.ElementsMatch(t, hookReq.GrantedScopes, expectedGrantedScopes)
 							require.ElementsMatch(t, hookReq.GrantedAudience, []string{})
 							require.Equal(t, hookReq.ClientID, oauthConfig.ClientID)
+							require.NotEmpty(t, hookReq.Session)
```

Comment:
Is there any better way to assert these objects? I would like to assert more things, hopefully, without giving each one its own assert.

Reply:
Try `github.com/ory/x/assertx.EqualAsJson` or you can also create a snapshot using `github.com/ory/x/snapshotx`